packages/jsapi-utils/src/useTable.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Column, Row, Table } from '@deephaven/jsapi-shim';
+import Log from '@deephaven/log';
+import useTableListener from './useTableListener';
+
+const log = Log.module('useTable');
+
+const useTable = (
+  table: Table | undefined,
+  columnNames: string[],
+  firstRow: number,
+  lastRow: number
```

Comment:
I think params should be flipped, so that `columnNames` is optional - if it's undefined, just get _all_ the columns.